APIMatic.Core/Utilities/CoreHelper.cs

```diff
@@ -567,31 +567,55 @@ private static void HandleCustomType(List<KeyValuePair<string, object>> processe
 
         private static void HandlePrimitiveTypes(List<KeyValuePair<string, object>> processedParameters, KeyValuePair<string, object> kvp)
         {
-            if (kvp.Value is IList)
+            if (kvp.Value is IList list)
             {
-                var list = kvp.Value as IList;
+                HandleListParameter(processedParameters, kvp, list);
+            }
+            else if (kvp.Value is IDictionary dictionary)
+            {
+                HandleDictionaryParameter(processedParameters, kvp, dictionary);
+            }
+            else
```

Comment:
We can avoid using `else` and `else-if` blocks here by converting these to `if` blocks combined with return statement at the end of each

Reply:
Done